### DISCUSSION TITLE
No temp sensors on OpenBSD

### DISCUSSION THREAD

**Discussion by TheQue42** [nicolargo/glances#1877] [Q&A] [answered]
Hi, 

I know the docs mainly reference FreeBSD, and I am trying out glances on OpenBSD, but I'd still like to ask...

Is there anyway to get the temp sensor display on OpenBSD?

If found various discussions on pysensors, which I seem to have issues with getting to install/run/find on OpenBSD 6.8, but I've a bit lost whether this is supposed to be possible at all?

The actual values ARE available, at:

```
#> sysctl hw.sensors
hw.sensors.cpu0.temp0=41.00 degC
hw.sensors.acpitz0.temp0=27.80 degC (zone temperature)
hw.sensors.acpitz1.temp0=29.80 degC (zone temperature)
hw.sensors.pchtemp0.temp0=40.00 degC
```

**Comment by nicolargo** [comment, marked as answer]
Hi @TheQue42 

the question of OpenBSD sensors support is not about Glances but about PsUtil. Glances uses PsUtil to grab the sensors stats.

Can you try this on your OpenBSD computer:

    # python
    > import psutil
    > psutil.sensors_temperatures()
    > psutil.sensors_fans()
    > psutil.sensors_batterry()

Note: you need PsUtil version 5.5.0 or higher.

If it is not working, you can ask an issue (enhancement request) to the PsUtil team ==> https://github.com/giampaolo/psutil/issues

**Reply by TheQue42** [reply]
I did suspect something like that, but since all dependencies from requirements.txt was satisfied, I was hoping it was something easier :-) 

And yes...psutil is where it fails :-( 

psutil.sensors_temperatures() returns: 

`AttributeError: module 'psutil' has no attribute 'sensors_temperatures'`

So I guess I'll have to try getting psutil updates with obsd support..

**Reply by TheQue42** [reply]
Although, now that read the psutil documentation is says: 

> If sensors are not supported by the OS an empty dict is returned, 

which does not seem to be the case...even though i am running psutil 5.8
